src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/VerifyCommand.cs

```diff
@@ -0,0 +1,82 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Commands;
+using NuGet.Common;
+using static NuGet.Commands.VerifyArgs;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal partial class VerifyCommand
+    {
+        internal static void Register(CommandLineApplication app,
+                              Func<ILogger> getLogger,
+                              Action<LogLevel> setLogLevel,
+                              Func<IVerifyCommandRunner> getCommandRunner)
+        {
+            app.Command("verify", verifyCmd =>
+            {
+                CommandArgument packagePaths = verifyCmd.Argument(
+                    "<package-paths>",
+                    Strings.VerifyCommandPackagePathDescription,
+                    multipleValues: true);
+
+                CommandOption all = verifyCmd.Option(
+                    "--all",
+                    Strings.VerifyCommandAllDescription,
+                    CommandOptionType.NoValue);
+
+                CommandOption fingerPrint = verifyCmd.Option(
+                    "--certificate-fingerprint",
+                    Strings.VerifyCommandCertificateFingerprintDescription,
+                    CommandOptionType.MultipleValue);
+
+                CommandOption verbosity = verifyCmd.Option(
+                    "-v|--verbosity",
+                    Strings.Verbosity_Description,
+                    CommandOptionType.SingleValue);
+
+                verifyCmd.HelpOption("-h|--help");
+                verifyCmd.Description = Strings.VerifyCommandDescription;
+
+                verifyCmd.OnExecute(async () =>
+                {
+                    ValidatePackagesPath(packagePaths);
+
+                    VerifyArgs args = new VerifyArgs();
+#if IS_DESKTOP
+                    args.PackagePath = packagePaths.Value;
+#else
+                    args.PackagePaths = packagePaths.Values;
+#endif
+                    args.Verifications = all.HasValue() ?
+                        new List<Verification>() { Verification.All } :
+                        new List<Verification>() { Verification.Signatures };
+                    args.CertificateFingerprint = fingerPrint.Values;
+                    args.Logger = getLogger();
+                    setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+                    var runner = getCommandRunner();
+                    var verifyTask = runner.ExecuteCommandAsync(args);
+                    await verifyTask;
+
+                    return verifyTask.Result;
+                });
+            });
+        }
+        private static void ValidatePackagesPath(CommandArgument argument)
+        {
+            if (argument.Values.Count == 0 ||
+                argument.Values.Any<string>(packagePath => string.IsNullOrEmpty(packagePath)))
```

Comment:
I added `IS_DESKTOP` symbol only for compiling the source code on `net472` platform because [`PackagePaths`](https://github.com/NuGet/NuGet.Client/blob/dev-kartheep-ms-dotnet-nuget-verify/src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyArgs.cs#L28-L38) is only available in `netcore` platforms. I can add a null check for `net472` platform. Please correct me if I am wrong, do we run `NuGet.CommandLine.Xplat dll` on `net472` platform?

Reply:
I think you're right. 
No need to add IS_DESKTOP for VerifyCommand under NuGet.CommandLine.XPlat.
`NuGet.Client\src\NuGet.Clients\NuGet.CommandLine\Commands\VerifyCommand.cs` works for net472  and `NuGet.Client\src\NuGet.Core\NuGet.CommandLine.XPlat\Commands\Signing\VerifyCommand.cs` works for netcore, and both of them call `NuGet.Client\src\NuGet.Core\NuGet.Commands\VerifyCommand\VerifyCommandRunner.cs`

Reply:
Both `nuget.exe verify` command and `dotnet nuget verify` command share [`VerifyArgs`](https://github.com/NuGet/NuGet.Client/blob/dev-kartheep-ms-dotnet-nuget-verify/src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyArgs.cs) as well.

Reply:
resolving this conversation because compilation symbol is removed from this file.